app/services/external_payment_processor.rb

```diff
@@ -2,10 +2,25 @@ class ExternalPaymentProcessor
   require "#{Rails.root}/lib/stripe_payment_processor.rb"
 
   attr_accessor :processed, :error
-
+  
+  #creates a new instance of ExternalPaymentProcessor that sets either processed as the successful response or error as the error message
+  #based on whether the response from the payment processor returns an id.
   def self.create_payment_process(options={})
-    response = payment_processor.new(options).process_card
-    response == true ? new(processed: response) : new(error: response)
+    payment_response = payment_processor.new(options).process_card
+    create_new_instance_from_response(payment_response)
```

Comment:
Hm, I feel this is a bit too much. Creating a new instance is just calling `.new` - so `create_new_instance_...` is in my opinion taxing the reader.
To me much simple way would be a method something like - `#processed?` which you could call on response.

```ruby
if payment_response.processed?
  new(processed: payment_response.success)
else
  new(error: payment_response.error_message)
end
```